### ISSUE TITLE
Deterministic output for SAC policy

### ISSUE THREAD

**Issue by DDDDYZ** [thu-ml/tianshou#233]
When I ran the example code, I found that even in the testing session, the SAC strategy is still a random policy. Shouldn't the mean value be output as a deterministic policy during the training?

**Comment by DDDDYZ** [comment]
When I ran the code in  https://github.com/thu-ml/tianshou/blob/master/examples/box2d/mcc_sac.py, the policy outputs random values during test. And I don't know how to determine the states of policies in tianshou. Should we use stochastic policies or the deterministic policies during tests?

**Comment by Trinkle23897** [comment]
What's the output and log?

**Comment by DDDDYZ** [comment]
I think in SAC algorithms, the entropy of policies increases during the training process, as a result, the test result will be unstable, which is exactly what I find during test. So I rewrote the collect method of Collector for test. The code is below.
Original version:
```python
            # calculate the next action
            if no_grad:
                with torch.no_grad():  # faster than retain_grad version
                    result = self.policy(self.data, last_state)          
            else:
                result = self.policy(self.data, last_state)
```
My version：
```python
            # calculate the next action
            if no_grad:
                with torch.no_grad():  # faster than retain_grad version
                    result = self.policy(self.data, last_state)
                    result.act = torch.tanh(result.logits[0])
            else:
                result = self.policy(self.data, last_state)
                result.act = torch.tanh(result.logits[0])
```
And I found the results became better.

**Comment by Trinkle23897** [comment]
But in fact, the tanh activation has been added into the network:
https://github.com/thu-ml/tianshou/blob/710966eda7308fe3d17e08cd9a2676f350b9b5e6/tianshou/utils/net/continuous.py#L110-L111
here, `not self._unbounded` is True, so in your pipeline there are two tanh activations.

**Comment by DDDDYZ** [comment]
In fact, I refer to the code in https://github.com/thu-ml/tianshou/blob/710966eda7308fe3d17e08cd9a2676f350b9b5e6/tianshou/policy/modelfree/sac.py#L108-L129
I think the point is whether to use deterministic policies or random policies when testing.

**Comment by DDDDYZ** [comment]
另外，我先用中文哈，为了方便，如果讨论出结果，我可以全部改成英文的

**Comment by Trinkle23897** [comment]
Yes, you are right. In fact, SAC paper demonstrates that the deterministic version is better than the stochastics version. This feature will be further written in tianshou.
(from @danagi)

**Comment by Trinkle23897** [comment]
> But in fact, the tanh activation has been added into the network:
> https://github.com/thu-ml/tianshou/blob/710966eda7308fe3d17e08cd9a2676f350b9b5e6/tianshou/utils/net/continuous.py#L110-L111
> 
> 
> here, `not self._unbounded` is True, so in your pipeline there are two tanh activations.

My bad, actually I'm wrong. There's no two tanh activation in SAC in any approach and I fix this issue in #235
